### PR TITLE
Fix Epidemic Sound SFX unsupported url

### LINKED ISSUE #10435: Epidemic Sound SFX new update unsupported URL
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm requesting a site-specific feature
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

_No response_

### Example URLs

https://www.epidemicsound.com/sound-effects/tracks/2f02f54b-9faa-4daf-abac-1cfe9e9cef69/

### Provide a description that is worded well enough to be understood

Epidemic sound had an update with their sound effects where the url are of a different pattern than the previous one. (i.e. _/sound-effects/track/_ instead of just _/track_)

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', 'https://www.epidemicsound.com/sound-effects/tracks/2f02f54b-9faa-4daf-abac-1cfe9e9cef69/']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version master@2024.07.11.172254 from yt-dlp/yt-dlp-master-builds [cc1a3098c] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.22631-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg 2023-06-11-git-09621fd7d9-full_build-www.gyan.dev (setts), ffprobe 2023-06-11-git-09621fd7d9-full_build-www.gyan.dev
[debug] Optional libraries: Cryptodome-3.20.0, brotli-1.1.0, certifi-2024.07.04, curl_cffi-0.7.0, mutagen-1.47.0, requests-2.32.3, sqlite3-3.35.5, urllib3-2.2.2, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets, curl_cffi
[debug] Loaded 1834 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-master-builds/releases/latest

Latest version: master@2024.07.11.172254 from yt-dlp/yt-dlp-master-builds
yt-dlp is up to date (master@2024.07.11.172254 from yt-dlp/yt-dlp-master-builds)
[generic] Extracting URL: https://www.epidemicsound.com/sound-effects/tracks/2f02f54b-9faa-4daf-abac-1cfe9e9cef69/
[generic] 2f02f54b-9faa-4daf-abac-1cfe9e9cef69: Downloading webpage
WARNING: [generic] Falling back on generic information extractor
[generic] 2f02f54b-9faa-4daf-abac-1cfe9e9cef69: Extracting information
[debug] Looking for embeds
ERROR: Unsupported URL: https://www.epidemicsound.com/sound-effects/tracks/2f02f54b-9faa-4daf-abac-1cfe9e9cef69/
Traceback (most recent call last):
  File "yt_dlp\YoutubeDL.py", line 1622, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1757, in __extract_info
  File "yt_dlp\extractor\common.py", line 740, in extract
  File "yt_dlp\extractor\generic.py", line 2521, in _real_extract
yt_dlp.utils.UnsupportedError: Unsupported URL: https://www.epidemicsound.com/sound-effects/tracks/2f02f54b-9faa-4daf-abac-1cfe9e9cef69/
```

### PR DESCRIPTION
**Pull request by iancmy**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

Fixed Epidemic Sound SFX unsupported url pattern (https://www.epidemicsound.com/sound-effects/tracks/2f02f54b-9faa-4daf-abac-1cfe9e9cef69/)

Fixes #10435

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Review by bashonly — CHANGES_REQUESTED** [review]
Thanks for the PR. Please see review comments, and also please add a test for the new url format

**Review by iancmy — COMMENTED** [review]
Thanks for the review! @bashonly 
- Implemented changes based on review
- Added test for the new url format

File: yt_dlp/extractor/epidemicsound.py [hunk 1]
```diff
@@ -77,8 +77,9 @@ def _epidemic_fmt_or_none(f):
         return f
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
-        json_data = self._download_json(f'https://www.epidemicsound.com/json/track/{video_id}', video_id)
+        video_id, is_sfx = self._match_valid_url(url).group('id', 'sfx')
+
+        json_data = self._download_json(join_nonempty('https://www.epidemicsound.com/json/track', is_sfx and 'kosmos-id', video_id, delim='/'), video_id)
```
Comment (bashonly) [inline yt_dlp/extractor/epidemicsound.py:82]: line too long
```suggestion
        json_data = self._download_json(join_nonempty(
            'https://www.epidemicsound.com/json/track'
            is_sfx and 'kosmos-id', video_id, delim='/'), video_id)
```

**Review by bashonly — CHANGES_REQUESTED** [review]

File: yt_dlp/extractor/epidemicsound.py [hunk 2]
```diff
@@ -77,8 +76,10 @@ def _epidemic_fmt_or_none(f):
         return f
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
-        json_data = self._download_json(f'https://www.epidemicsound.com/json/track/{video_id}', video_id)
+        video_id, is_sfx = self._match_valid_url(url).group('id', 'sfx')
+        json_data = self._download_json(join_nonempty(
+            'https://www.epidemicsound.com/json/track'
```
Comment (bashonly) [inline yt_dlp/extractor/epidemicsound.py:81]:
```suggestion
            'https://www.epidemicsound.com/json/track',
```

File: yt_dlp/extractor/epidemicsound.py [hunk 3]
```diff
@@ -13,24 +14,8 @@
...
-            'title': 'Door Knock Door 1',
-            'alt_title': 'Door Knock Door 1',
-            'tags': ['foley', 'door', 'knock', 'glass', 'window', 'glass door knock'],
-            'categories': ['Misc. Door'],
-            'duration': 1,
-            'thumbnail': 'https://cdn.epidemicsound.com/curation-assets/commercial-release-cover-images/default-sfx/3000x3000.jpg',
-            'timestamp': 1415320353,
-            'upload_date': '20141107',
-        },
-    }, {
```
[the remark this answers is not in the document]
Comment (bashonly) [reply yt_dlp/extractor/epidemicsound.py:33]: test passes too

**Review by bashonly — APPROVED** [review]

**Pull request merged**